scripts/upgrade_kernel.sh

```diff
@@ -5,9 +5,13 @@ set -o nounset
 set -o errexit
 
 if [[ -z "$KERNEL_VERSION" ]]; then
-  KERNEL_VERSION=5.4
-
-  echo "kernel_version is unset. Setting to $KERNEL_VERSION"
+  if vercmp "$KUBERNETES_VERSION" gteq "1.24.0"; then
+    KERNEL_VERSION=5.10
+    echo "kernel_version is unset. Setting to $KERNEL_VERSION based on Kubernetes version $KUBERNETES_VERSION."
+  else
+    KERNEL_VERSION=5.4
+    echo "kernel_version is unset. Setting to $KERNEL_VERSION."
+  fi
 fi
```

Comment:
🔽  would it be simpler like this? 

```
if [[ -z "$KERNEL_VERSION" ]]; then
  KERNEL_VERSION=5.4

  echo "kernel_version is unset. Setting to $KERNEL_VERSION"
  if vercmp "$KUBERNETES_VERSION" gteq "1.24.0"; then
    KERNEL_VERSION=5.10
  fi

  echo "kernel_version is unset. Setting to $KERNEL_VERSION based on Kubernetes version $KUBERNETES_VERSION."

fi
```

Reply:
I don't think we need both the `echo`-s, but it does make sense to log the k8s version either way